specs/PrintToPdf.md

```diff
@@ -0,0 +1,294 @@
+        printSettings.Orientation =
+            CoreWebView2PrintOrientation.Landscape;
+    }
+    string resultFilePath = await webView.CoreWebView2.PrintToPdfAsync(
+        "" /* use default path*/, printSettings);
+    MessageBox.Show(this, resultFilePath, "Print To PDF Completed");
+}
+```
+# Remarks
+
+# API Notes
+
+
+# API Details
+```
+interface ICoreWebView2_4;
+interface ICoreWebView2Environment5;
+interface ICoreWebView2PrintSettings;
+interface ICoreWebView2PrintToPdfCompletedHandler;
+
+
+/// The orientation for printing, used by the `Orientation` property on
+/// `ICoreWebView2SPrintSettings`. Currently only printing to PDF
+/// is supported.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_ORIENTATION
+{
+  /// Print the page(s) in portrait orientation.
+  COREWEBVIEW2_PRINT_ORIENTATION_PORTRAIT,
+
+  /// Print the page(s) in landscape orientation.
+  COREWEBVIEW2_PRINT_ORIENTATION_LANDSCAPE,
+} COREWEBVIEW2_PRINT_ORIENTATION;
+
+[uuid(9eae81c3-fe02-4084-b475-903a4ed9252e), object, pointer_default(unique)]
+interface ICoreWebView2_4 : IUnknown {
+  /// Print the current page to PDF asynchronously with the provided settings.
+  /// See `ICoreWebView2PrintSettings` for description of settings.
+  ///
+  /// Use `resultFilePath` to specify the path to the PDF file. The host should
+  /// provide an absolute path, including file name. If the path
+  /// points to an existing file, the file will be overwritten. If the directory
+  /// does not exist, it is created. If `resultFilePath` is empty, the default
+  /// path will be used.
+  ///
+  /// When `PrintToPdf` completes, the `ICoreWebView2PrintToPdfCompletedHandler`
+  /// is invoked.
+  ///
+  /// \snippet FileComponent.cpp PrintToPdf
+  HRESULT PrintToPdf(
+    [in] LPCWSTR resultFilePath,
+    [in] ICoreWebView2PrintSettings* printSettings,
+    [in] ICoreWebView2PrintToPdfCompletedHandler* handler);
+}
+
+/// Receives the result of the `PrintToPdf` method. The `resultFilePath`
+/// contains the final path of the PDF file.
+[uuid(4808ac58-c372-4d3d-be5e-900df2593835), object, pointer_default(unique)]
+interface ICoreWebView2PrintToPdfCompletedHandler : IUnknown {
+
+  /// Provides the result of the corresponding asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode, [in] LPCWSTR resultFilePath);
+}
+
+/// Settings used by the `PrintToPdf` method. Other programmatic printing is not
+/// currently supported.
+[uuid(30964a64-2e92-4604-9eaa-3afcd3a015dc), object, pointer_default(unique)]
+interface ICoreWebView2PrintSettings : IUnknown {
+
+  /// The orientation can be portrait or landscape. The default orientation is
+  /// portrait. See `COREWEBVIEW2_PRINT_ORIENTATION`.
+  [propget] HRESULT Orientation(
+    [out, retval] COREWEBVIEW2_PRINT_ORIENTATION* orientation);
+
+  /// Sets the `Orientation` property.
+  [propput] HRESULT Orientation(
+      [in] COREWEBVIEW2_PRINT_ORIENTATION orientation);
+
+  /// The scale factor is a value between 0.1-2.0. The default is 1.0.
```

Comment:
For all the doubles here, please specify the valid range and what happens if a value outside the range is specified (should match what we do for invalid range specified for our existing controller scale property)